python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -868,6 +868,36 @@ def check_raylet_healthy():
             sleep(1)
 
 
+def test_session_name(ray_start_cluster):
+    # Kill GCS and check that raylets kill themselves when not backed by Redis,
+    # and stay alive when backed by Redis.
+    # Raylets should kill themselves due to cluster ID mismatch in the
+    # non-persisted case.
+    cluster = ray_start_cluster
+    cluster.add_node()
+    cluster.wait_for_nodes()
+
+    head_node = cluster.head_node
+    session_dir = head_node.get_session_dir_path()
```

Comment:
CAn you also submit a driver and check if their session name is correct? 

```
import ray
ray.init()
# Make sure this is correct
ray._private.worker._global_node.get_session_dir_path()
```

We need this

1. Start the first head node. Submit a driver to a head node and get a session dir
2. Start the worker node. Submit a driver to "worker node" and get a session dir.
3. Kill head node. Restart head node.
4. Submit a driver to a head node and get a session dir
5. Submit a driver to a worker node and get a session dir. 

This could also be done in a similar way as `test_gcs_ha_e2e.py` (it starts head / worker in docker containers, so it is easy to test it)

Reply:
Maybe add the same suite of tests in this commit; https://github.com/ray-project/ray/pull/37644/commits/748ddf8791d9284209305e492716f5ea60fe0c46?